presto-main/src/main/java/com/facebook/presto/sql/planner/plan/JoinNode.java

```diff
@@ -132,6 +132,18 @@ public JoinNode flipChildren()
                 distributionType);
     }
 
+    public static boolean mustPartition(JoinNode.Type type)
```

Comment:
you can make both `mustPartition` and `mustReplicate` non static, so you don't have to pass the parameters

Reply:
no because it's used in predicate push down to determine if it *can* create that kind a node with those features.